stdlib/public/core/HashTable.swift

```diff
@@ -70,22 +70,23 @@ extension _HashTable: Sendable {}
 
 extension _HashTable {
   /// The inverse of the maximum hash table load factor.
-  private static var maxLoadFactor: Double {
-    @inline(__always) get { return 3 / 4 }
+  private static var maxLoadFactor: (numerator: Int, denominator: Int) {
+    @inline(__always) get { return (numerator: 3, denominator: 4) } // 3/4
   }
 
   internal static func capacity(forScale scale: Int8) -> Int {
     let bucketCount = (1 as Int) &<< scale
-    return Int(Double(bucketCount) * maxLoadFactor)
+    return bucketCount * maxLoadFactor.numerator / maxLoadFactor.denominator
```

Comment:
For 32b and smaller systems, we plausibly might care about intermediate overflow here.

Reply:
Any suggestions? We could ifdef this change to apply just to `#if $Embedded`, perhaps it would be reasonable to assume that 32-bit *embedded* systems will never have use a hashtable that's on the order of ~1 billion entries?

Or would using 64-bit arithmetics even on 32-bit systems be reasonable?

Reply:
Do we expect to want to tweak the load factor, @lorentey?

Reply:
Aha, so would it be reasonable to replace the multiply-then-divide (possible overflow) with divide-then-multiply (no overflow) plus handling the remainder separately?

Reply:
I wonder if there is a way we can avoid division and use shifting instead? cortex M0 is kinda poor perf for division no?

Reply:
For the hard-coded divide by 4 here that works, if we need to be more general we could do a libdivide-style integer reciprocal multiplication.